src/main/webapp/js/index.js

```diff
@@ -12,8 +12,94 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/* Date range picker handler:
- * https://www.daterangepicker.com/
+
+/** Fetch receipts from the server and add them to the DOM. */
+async function searchReceipts() {
+  const label = document.getElementById('search-input').value;
+  const response = await fetch(`/search-receipts?label=${label}`);
+  const receipts = await response.json();
+
+  clearExistingReceipts();
+  displayReceipts(label, receipts);
+}
+
+/**
+ * Clear out existing receipts display.
+ *
+ * Note: 3 childNodes make up a receipt card.
+ * The first 3 nodes are the template receipt card and each subsequent
+ * 3 are clones of the template. The template must never be deleted or
+ * it'll be unusable in the future.
+ */
+function clearExistingReceipts() {
+  const receiptsDisplay = document.getElementById('receipts-display');
+  const numChildNodes = receiptsDisplay.childNodes.length;
+
+  // Make sure not to delete first 3 childNodes
+  for (let i = numChildNodes; i > 3; i--) {
+    receiptsDisplay.removeChild(receiptsDisplay.lastChild);
+  }
+}
+
+/**
+ * Populate receipt display with newly queried receipts.
+ * @param {string} label User-entered label.
+ * @param {JSON Object} receipts Receipts returned from search query.
+ */
+function displayReceipts(label, receipts) {
+  // If no receipts returned, display an error message. Else, display receipts.
+  if (Object.keys(receipts).length == 0) {
+    createErrorMessageElement(label);
+  } else {
+    receipts.forEach((receipt) => {
+      createReceiptCardElement(receipt);
+    });
+  }
+}
+
+/**
+ * Create receipt card based on existing HTML template.
+ * This card displays transaction date, store name, trasaction total,
+ * categories, receipt photo, and view/edit/delete buttons.
+ * @param {Receipt} receipt A Receipt object.
+ */
+function createReceiptCardElement(receipt) {
+  const categories = Array.from(receipt.categories);
+
+  // Fill in template fields with correct information.
+  const templateScope = document.querySelector('#receipt-card-template');
+  templateScope.content.querySelector('#timestamp').innerText =
+      receipt.timestamp;
+  templateScope.content.querySelector('#store-name').innerText = receipt.store;
+  templateScope.content.querySelector('#total').innerText =
+      'Total: $' + receipt.price;
+  templateScope.content.querySelector('#c1').innerText = categories[0];
+  templateScope.content.querySelector('#c2').innerText = categories[1];
+  templateScope.content.querySelector('#c3').innerText = categories[2];
+  templateScope.content.querySelector('img').src = receipt.imageUrl;
+
+  // Clone and attach receipt card to parent div.
+  const receiptCardClone = document.importNode(templateScope.content, true);
```

Comment:
Based on [1] importNode is used to import a node from another document (like an iframe). Although this will work, I think it's more correct to use `<node>.cloneNode()` within the same document.

[1] https://www.w3schools.com/jsref/met_document_importnode.asp